torch/utils/throughput_benchmark.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import torch._C
+
+class ThroughputBenchmark(object):
+    '''
+    This class is a wrapper around a c++ component throughput_benchmark::ThroughputBenchmark
+    responsible for executing a PyTorch module (nn.Module or ScriptModule)
+    under an inference server like load. It can emulate multiple calling threads
+    to a single module provided. In the future we plan to enhance this component
+    to support inter and intra-op parallelism as well as multiple models
+    running in a single process.
+
+    Please note that even though nn.Module is support, it might incur an overhead
+    from the need to hold GIL every time we execute Python code or pass around
+    inputs as Python objects. As soon as you have a ScriptModule version of your
+    model for inference deployment it is better to switch to using it in this
+    benchmark.
+
+    Example usage:
+
+        from torch.utils import ThroughputBenchmark
+
+        bench = ThroughputBenchmark(my_module)
+
+        for input in inputs:
+            # Both args and kwargs work, same as any PyTorch Module / ScriptModule
+            bench.add_input(input[0], x2=input[1])
+
+        stats = bench.benchmark(
+            num_calling_threads=4,
+            num_warmup_iters = 100,
+            num_iters = 1000,
+        )
+
+        print("Avg latency (ms): {}".format(stats.latency_avg_ms))
+        print("Number of iterations: {}".format(stats.num_iters))
+
+    '''
+
+    def __init__(self, module):
+        if isinstance(module, torch.jit.ScriptModule):
+            self._benchmark = torch._C.ThroughputBenchmark(module._c)
+        else:
+            self._benchmark = torch._C.ThroughputBenchmark(module)
+
+    def run_once(self, *args, **kwargs):
+        '''
+        Given input id (input_idx) run benchmark once and return prediction.
+        This is useful for testing that benchmark actually runs the module you
+        want it to run. input_idx here is an index into inputs array populated
+        by calling add_input() method.
+        '''
+        return self._benchmark.run_once(*args, **kwargs)
+
+    def add_input(self, *args, **kwargs):
+        '''
+        Store a single input to a module into the benchmark memory
+        '''
+        self._benchmark.add_input(*args, **kwargs)
+
+    def benchmark(self, num_calling_threads=1, num_warmup_iters=10, num_iters=100):
+        '''
+        num_calling_threads Calling threads are those threads that are calling
+        into a module in parallel.
+
+        num_warmup_iters Warmup iters are used to make sure we run a module
```

Comment:
the docstrings can be improved by following the Args:: format, with the added benefit that they'll be rendered beautifully in the final docs:
https://github.com/pytorch/pytorch/blob/master/torch/_tensor_docs.py#L584-L588
https://pytorch.org/docs/stable/tensors.html#torch.Tensor.copy_